spectator-reg-atlas/src/main/java/com/netflix/spectator/atlas/impl/EvalPayload.java

```diff
@@ -59,6 +60,31 @@ public long getTimestamp() {
     return messages;
   }
 
+  /**
+   * Break the payload down to a set of batches to limit the size of requests going to the
+   * service.
+   *
+   * @param batchSize
+   *     Size of the metric batches to create.
+   * @return
+   *     List of payloads that have at most {@code batchSize} metrics per payload.
+   */
+  public List<EvalPayload> toBatches(int batchSize) {
+    int size = metrics.size();
+    if (size <= batchSize) {
+      return Collections.singletonList(this);
+    } else {
+      List<EvalPayload> payloads = new ArrayList<>(size / batchSize + 2);
```

Comment:
Not +1 here?

Reply:
Good catch, the +2 was from an earlier attempt where I was looking at splitting the messages as well. Decided it wasn't worth the complexity right now with current usage.